### ISSUE TITLE
literalinclude line numbers not shown if invalid code

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#845]
I put :linenos: on a literalinclude directive and was puzzled because only some code blocks got highlighted and had line numbers displayed.  It turns out that highlight_block() in sphinx/highlighting.py does a trial parse of the code block when the language is "python" and displays an unhighlighted version when the code can't be parsed.  

I think this should be documented in the description of :literalinclude:.  Draft text:

```
If the language is "python" or "py", code highlighting requires
that the code must be either a console transcript or syntactically
valid Python code (parsed by the version of Python that's running
Sphinx).  
Syntactically-incorrect Python code will not be highlighted,
have line numbers displayed, or lines emphasized.  Python 3.x
code, declared by setting language to 'python3' or 'py3', will not
be checked for correct syntax.
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/845
- Originally reported by: Andrew Kuchling
- Originally created at: 2011-12-30T23:24:49.031

**Comment by shimizukawa** [comment]
_From Andrew Kuchling on 2012-01-24 22:51:50+00:00_

Issue #695 has a patch that changes the highlighting behaviour.  If it's accepted, the suggested text might be incorrect (depending on what gets committed).

**Comment by shimizukawa** [comment]
_From Anonymous on 2012-05-23 22:15:59+00:00_

fwiw I just ran into this bug. it would be nice if this was documented or at least emitted a warning.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-11 09:00:14+00:00_

Closes #845: In code blocks, when the selected lexer fails, display line numbers nevertheless if configured.

→ <<cset b29226a0609d5803afd51c01f6d522bd30592dbf>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-11 09:00:32+00:00_

Should be really fixed now, thanks for the report.
